src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.TextChunk.cs

```diff
@@ -32,7 +32,10 @@ private struct TextChunk : IDisposable
             /// </summary>
             public TemporaryArray<TextSpan> PatternHumps;
 
-            public readonly WordSimilarityChecker? SimilarityChecker;
+            /// <summary>
+            /// Not readonly as this value caches data within it, and so it needs to be able to mutate.
+            /// </summary>
+            public WordSimilarityChecker SimilarityChecker;
```

Comment:
the alternative is to accept that thsi caching optimization was never working, remove it entirely, and make all this stuff readonly.